### PR TITLE
Implement ChatWindow and ChatMessages components

### PR DESCRIPTION
**Pull request by Simhanischal**
- Implemented ChatWindow and ChatMessages components.
- Fixed formatting for few components.

Screenshots:

image

https://github.com/Simhanischal/Mini-ChatGPT/assets/33859648/c412c317-3240-4502-abd7-4187fa8f6e02

### CODE REVIEW

File: src/assets/stylesheets/chat/chatBanner.scss [hunk 1]
```diff
@@ -16,7 +16,7 @@
   border-radius: 50%;
   background-color: #ffffff;
   padding: 2px 4px;
-  margin-left: 20px;
+  margin-left: 10px;
```
Comment (Simhanischal) [inline src/assets/stylesheets/chat/chatBanner.scss:19]: Reduced font size as 20px looked like it left a big gap.

File: src/assets/stylesheets/chat/typingIndicator.scss [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.container {
+.typing-indicator-container {
```
Comment (Simhanischal) [inline src/assets/stylesheets/chat/typingIndicator.scss:1]: Changed class names in few components as we had the same className for parent and its child component which resulted in unpredictable styling behaviour.

File: src/components/chat/chatBanner/ChatBanner.tsx [hunk 3]
```diff
@@ -1,26 +1,24 @@
 import Typography from "@material-ui/core/Typography";
 import { green } from "@material-ui/core/colors";
 import Image from "../../../assets/images/logo192.png";
-import '../../../assets/stylesheets/chat/chatBanner.scss'
+import "../../../assets/stylesheets/chat/chatBanner.scss";
 
 const ChatBanner = () => {
   return (
-    <>
```
Comment (Simhanischal) [inline src/components/chat/chatBanner/ChatBanner.tsx:8]: Removed unnecessary Fragment

File: src/components/chat/chatInput/ChatInput.tsx [hunk 4]
```diff
@@ -23,24 +23,21 @@ const ChatInput = (props: ChatInputProps) => {
   });
 
   return (
-    <>
```
Comment (Simhanischal) [inline src/components/chat/chatInput/ChatInput.tsx:26]: Removed unnecessary Fragment

File: src/components/chat/Chat.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+interface Message {
+  type: string;
+  message: string;
+  datetime: string;
+}
+
+interface ChatProps {
+  messages: Array<Message>;
+  setMessages: React.Dispatch<React.SetStateAction<Array<Message>>>;
```
Comment (medhapatil12) [inline src/components/chat/Chat.tsx:17]: 👌🏽

File: src/components/chat/Chat.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+  //repalce the placeholder method definition with the actual API call
+  const fetchBotReply = () => {
+    setTimeout(() => {
+      setMessages((oldMessages: Array<Message>) => [
+        ...oldMessages,
+        {
+          type: "bot",
+          //default message
+          message:
+            "Sorry, I dont understand that! Please connect me to a ChatBot API to get proper responses! Have a good day!",
```
Comment (medhapatil12) [inline src/components/chat/Chat.tsx]: nit: don't 😬

Comment (Simhanischal) [reply src/components/chat/Chat.tsx]: addressed: https://github.com/Simhanischal/Mini-ChatGPT/commit/8b85bc79056ed6863cd668cf92cc119a720c234d

**Review by medhapatil12 — APPROVED** [review]

**Pull request merged**